pkg/match/payload/json_comparator.go

```diff
@@ -8,7 +8,34 @@ import (
 type JSONComparator struct {
 }
 
+func isArray(s1 string) bool {
+	return len(s1) > 0 && s1[0] == '['
+}
+
+func (jc *JSONComparator) compareArray(s1, s2 string) bool {
+
+	var o1 []interface{}
+	var o2 []interface{}
+
+	var err error
+	err = json.Unmarshal([]byte(s1), &o1)
+	if err != nil {
+		return false
+	}
+	err = json.Unmarshal([]byte(s2), &o2)
+	if err != nil {
+		return false
+	}
+
+	return reflect.DeepEqual(o1, o2)
+}
+
 func (jc *JSONComparator) Compare(s1, s2 string) bool {
+
+	if isArray(s1) || isArray(s2) {
+		return jc.compareArray(s1, s2)
+	}
+
```

Comment:
Nice feature!

Just a minor style comment: the `compareArray` method is practically the same, only the temporal variables types change.

What about doing something like this? (untested)
```go
if isArray(s1) || isArray(s2) {
	var o1 []interface{}
	var o2 []interface{}
} else {
	var o1 interface{}
	var o2 interface{}
}
```

Reply:
I tried...  and it would work on other languages but not on go.. that's why we had to duplicate everything..  we are trying to extract the common part but it's not that simple as we need to pass the type of the objects (which will be different for each case) to the function..